### PR TITLE
[int] SQL escaping tidyups

### PR DESCRIPTION
**Pull request by sfayer**
Hi,

Here is my suggested tidy-up for allowing the DATE & RAND functions through `quotedList` by generalising the existing date function support.

I've also added a test-case for the escapeString None->NULL special case that was fixed earlier in the week.

Regards,
Simon

BEGINRELEASENOTES
*Core
CHANGE: SQL escaping tidyups
ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/Core/Utilities/MySQL.py [hunk 1]
```diff
@@ -1306,7 +1329,7 @@ def getCounters(
             # self.log.debug('getCounters:', error)
             return S_ERROR(DErrno.EMYSQL, error)
 
-        attrNames = _quotedList(attrList, allowDate=True)
```
Comment (fstagni) [inline src/DIRAC/Core/Utilities/MySQL.py:1309]: This looks to be the only usage in DIRAC. There's no usage in LHCbDIRAC, so maybe drop it altogether?

Comment (sfayer) [reply src/DIRAC/Core/Utilities/MySQL.py:1309]: Sure, I wasn't sure if anyone else was using it (but suspected they probably weren't): I've dropped it now.

**Review by fstagni — APPROVED** [review]
